eventstream.go

```diff
@@ -44,36 +54,40 @@ type Event struct {
 	UserID    string                 `json:"userId"`
 	SessionID string                 `json:"sessionId"`
 	Timestamp string                 `json:"timestamp"`
-	Version   string                 `json:"version"`
+	Version   int                    `json:"version"`
 	Payload   map[string]interface{} `json:"payload"`
 }
 
 // BrokerConfig is custom configuration for message broker
 type BrokerConfig struct {
-	DialTimeout  time.Duration
-	ReadTimeout  time.Duration
-	WriteTimeout time.Duration
+	LogMode          level
+	StrictValidation bool
+	DialTimeout      time.Duration
+	ReadTimeout      time.Duration
+	WriteTimeout     time.Duration
 }
 
 // PublishBuilder defines the structure of message which is sent through message broker
 type PublishBuilder struct {
-	topic     []string
-	eventName string
-	namespace string
-	clientID  string
-	traceID   string
-	userID    string
-	sessionID string
-	version   string
-	payload   map[string]interface{}
-	ctx       context.Context
+	topic         []string
+	eventName     string
+	namespace     string
+	clientID      string
+	traceID       string
+	userID        string
+	sessionID     string
+	version       int
+	payload       map[string]interface{}
+	callbackError func(event *Event, err error)
```

Comment:
"errorCallback" is more grammatically correct.
what's the thought process of only calling callback for errors BTW?

Reply:
yes, its triggered when the publisher already retrying but the event still not published.

Reply:
so if it's still retrying, why already calling the error callback?

we can go either:
- notify on retry and failure, so that means we need to tell the caller if the publisher have given up or not
- notify on failure only, when the publisher have given up.

don't confuse the caller by calling error callback while it's still retrying.

Reply:
sorry, I means the callback will calling when the publisher have given up

Reply:
Ah ok, make sense